### PR TITLE
2096 Replace Standardization interpreter

### PR DESCRIPTION
**Pull request by Zejnilovic**
Draft PR of the implementation of standardization library.

Still need to delete a lot of code around other parts.

### CODE REVIEW

**Review by dk1844 — APPROVED** [review]
LGTM so far (just read the code)

File: spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala [hunk 1]
```diff
@@ -83,9 +83,10 @@ trait StandardizationExecution extends CommonJobExecution {
     // Add the raw format of the input file(s) to Atum's metadata
     Atum.setAdditionalInfo("raw_format" -> cmd.rawFormat)
 
-    val defaultTimeZoneForTimestamp = defaults.getDefaultTimestampTimeZone.getOrElse(spark.conf.get("spark.sql.session.timeZone"))
+    // TODO
```
[the remark this answers is not in the document]
Comment (Zejnilovic) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala:86]: This TODO came from my 1st misunderstanding of how it works.

File: spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala [hunk 2]
```diff
@@ -180,6 +177,7 @@ trait StandardizationExecution extends CommonJobExecution {
     val rawFs = preparationResult.pathCfg.raw.fileSystem
     val stdFs = preparationResult.pathCfg.standardization.fileSystem
 
+    // Todo for Standardization extraction
```
[the remark this answers is not in the document]
Comment (Zejnilovic) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala:180]: So this is more complicated. The todo was left there by a mistake. At that point, I just realized we would need much more extraction. And a lot of things are also for spark-commons, not just standardization

**Review by dk1844 — DISMISSED** [review]
LGTM (read the code; checked out and poked around)

File: spark-jobs/src/test/scala/za/co/absa/enceladus/standardization/samples/TestSamples.scala [hunk 3]
```diff
@@ -15,13 +15,14 @@
...
 package za.co.absa.enceladus.standardization.samples
 
+import za.co.absa.standardization.config.DefaultErrorCodesConfig
+
 import java.sql.Timestamp
 import java.sql.Date
 import java.math.BigDecimal
-import za.co.absa.enceladus.utils.error.ErrorMessage
-import java.time.LocalDate
+//import za.co.absa.enceladus.utils.error.ErrorMessage
```
Comment (dk1844) [inline spark-jobs/src/test/scala/za/co/absa/enceladus/standardization/samples/TestSamples.scala:23]: cleanup?

**Review by dk1844 — DISMISSED** [review]
Still LGTM (just read the code)

**Review by dk1844 — APPROVED** [review]

**Review by dk1844 — APPROVED** [review]
LGTM (just read the code)

**Comment by dk1844** [comment]
I have built both `develop` and this branch and ran a Standardization of the `mappingConformanceRule` dataset from Enceladus examples. Loaded the parquets and compared them to identical except for randomly generated `enceladus_record_id`. 

```bash
scala> val stddev = spark.read.parquet("""C:\test\mappingConformanceRule\std-develop\mappingConformanceRule\1\2020-03-23\v1""")
stddev: org.apache.spark.sql.DataFrame = [property: string, name: string ... 6 more fields]

scala> val std2096 = spark.read.parquet("""C:\test\mappingConformanceRule\std-2096\mappingConformanceRule\1\2020-03-23\v1""")
std2096: org.apache.spark.sql.DataFrame = [property: string, name: string ... 6 more fields]

scala> stddev.show(false)
+--------+-----------------+----------+---------------------------------------+------------------------+-----------------------------------------------------+------+------------------------------------+
|property|name             |location  |address                                |devices                 |accesses                                             |errCol|enceladus_record_id                 |
+--------+-----------------+----------+---------------------------------------+------------------------+-----------------------------------------------------+------+------------------------------------+
|user1   |Jindrich Sidlo   |internet  |[Happy Monday, Praha, [110 00, 120 00]]|[mobile phone, notebook]|[[library, 8:00 - 20:00], [rest room, 12:00 - 14:00]]|[]    |3aece17c-6b06-4557-b750-53d928a66a40|
|user2   |Martin Veselovsky|internet  |[DVTV, Praga, []]                      |[phone]                 |[[library, 9:00 - 19:00], [rest room, 11:00 - 15:00]]|[]    |47c5cb29-fb58-4257-ae84-80dc4b3b249e|
|user3   |Nora Fridrichova |television|[168Hours, Prag, [100 00, 160 00]]     |[fax, notebook]         |[[library, 7:00 - 15:00], [kitchen, 6:00 - 14:00]]   |[]    |075ce231-c63b-40fe-bb35-f31f17a9bf24|
+--------+-----------------+----------+---------------------------------------+------------------------+-----------------------------------------------------+------+------------------------------------+


scala> std2096.show(false)
+--------+-----------------+----------+---------------------------------------+------------------------+-----------------------------------------------------+------+------------------------------------+
|property|name             |location  |address                                |devices                 |accesses                                             |errCol|enceladus_record_id                 |
+--------+-----------------+----------+---------------------------------------+------------------------+-----------------------------------------------------+------+------------------------------------+
|user1   |Jindrich Sidlo   |internet  |[Happy Monday, Praha, [110 00, 120 00]]|[mobile phone, notebook]|[[library, 8:00 - 20:00], [rest room, 12:00 - 14:00]]|[]    |f36c49f9-7c1c-4a57-b06c-7d6a5a572ac9|
|user2   |Martin Veselovsky|internet  |[DVTV, Praga, []]                      |[phone]                 |[[library, 9:00 - 19:00], [rest room, 11:00 - 15:00]]|[]    |ad160668-73c6-45d2-98a8-f170f8849fe9|
|user3   |Nora Fridrichova |television|[168Hours, Prag, [100 00, 160 00]]     |[fax, notebook]         |[[library, 7:00 - 15:00], [kitchen, 6:00 - 14:00]]   |[]    |710d3cd3-fd0d-4360-b94e-0337eb4399b9|
+--------+-----------------+----------+---------------------------------------+------------------------+-----------------------------------------------------+------+------------------------------------+

scala> stddev.drop('enceladus_record_id).except(std2096.drop('enceladus_record_id)).isEmpty
res11: Boolean = true

scala> std2096.drop('enceladus_record_id).except(stddev.drop('enceladus_record_id)).isEmpty
res12: Boolean = true
```

Marking as tested.

**Pull request merged**